php-checks/src/main/java/org/sonar/php/checks/security/ClearTextProtocolsCheck.java

```diff
@@ -181,8 +242,84 @@ private static LaravelMailConfig of(ArrayInitializerTree tree) {
       return new LaravelMailConfig(transport, host, encryption);
     }
 
-    private boolean isClearText() {
-      return hasInsecureEncryption() && hasInsecureHost();
+    private boolean isSmtp() {
+      return transport != null && transport.is(Tree.Kind.REGULAR_STRING_LITERAL) && "smtp".equals(CheckUtils.trimQuotes((LiteralTree) transport));
+    }
+  }
+
+  private static class SwiftMailConfig extends MailConfig {
+    public SwiftMailConfig(@Nullable ExpressionTree host, @Nullable ExpressionTree encryption) {
+      super(host, encryption);
+    }
```

Comment:
If the cunstructor simply passes on arguments, then it is not needed, is it?

Reply:
Yes it is needed. Without it the code does not work.